### ISSUE TITLE
useIdleTimer: handler onIdle is not called on long timeout for mobile device if tab is inactive

### ISSUE THREAD

**Issue by AlexHuman** [SupremeTechnopriest/react-idle-timer#271]
### Affected Module

useIdleTimer, workerTimers

- `Version`: 5.4.0

### Describe the bug

Handler `onIdle` is not called on long timeout (i.e 10 minutes) for mobile device if tab is inactive, even if `workerTimers` are set as timers.

### To Reproduce

1. Setup `timers`, `timeout` and `throttle` for `useIdleTimer`:

```
useIdleTimer({
    element: document,
    timers: workerTimers,
    timeout: 10 * 60 * 1000, // 10 min
    throttle: 3000, // 3 sec
    onIdle,
    onAction,
  });
```

2. Open page and make actions
3. Change tab and wait 10min +
4. Return to the first tab

### Expected behavior

`onIdle` is called on time, duration between last action and onIdle call is about 10min.

### Actual behavior

`onIdle` is called only on tab activation.

### Screenshots

Time logs:
![1](https://user-images.githubusercontent.com/23736906/176437882-4ed28558-0208-4533-ae2f-26e13d098944.png)

### System Information
1) 
 - `OS`: Android 12
 - `Device`: Vivo v21
 - `Browser Vendor`: chrome
 - `Browser Version`: 102
 -  `Other info`: USB-debugging with desktop chrome v103
 - `Timeouts`:
     - 4.30 min: OK
     - 5 min: FAIL
     - 10 min: FAIL
  
2)
 - `OS`: Android 11
 - `Device`: Browserstack Samsung Galaxy s21
 - `Browser Vendor`: chrome
 - `Browser Version`: 100
 - `Timeouts`:
     - 10 min: FAIL

3)
 - `OS`: IOS 15.4
 - `Device`: Browserstack IPhone 11
 - `Browser Vendor`: safari
 - `Browser Version`: 604.1
 - `Timeouts`:
     - 10 min: OK
     
4)
 - `OS`: IOS 15.4
 - `Device`: Browserstack IPhone 11
 - `Browser Vendor`: chrome
 - `Browser Version`: 92
 - `Timeouts`:
     - 10 min: OK

### Additional context

1. Desktop browsers works well thanks to `workerTimers`.
2. The same issue appears for `onPrompt` handler and long `promptTimeout`.

**Comment by SupremeTechnopriest** [comment]
Thanks for the well reported issue.  Gonna have to do some research on this one.  The only reason I could think of that this would happen is android chrome is throttling worker execution when the page is backgrounded.  To my knowledge, that's not supposed to happen.

I have to find my old android device to debug this.  In the mean time can you confirm for me that this only happens with long timers?  Could you try setting the timeout to the following intervals and see when it starts being throttled:

- [ ] 1 Minute
- [ ] 3 Minutes
- [ ] 5 Minutes
- [ ] 7 Minutes

Knowing exactly when the throttling starts would help direct my research. 

Thanks!

**Comment by SupremeTechnopriest** [comment]
Yeah IOS on chrome is actually running webkit.  Safari does not support the actual chrome engine blink. So that makes sense.  This sounds like a bug in the chrome engine for mobile.  I think we are going to have to report it and wait for a fix from them, which is unfortunate...  Sounds like the workers stop executing when the app is backgrounded for more than 5 mins.

**Comment by ivansky** [comment]
Despite of the possible bug, there might be some mobile optimisations that just stops the whole page, timeouts, intervals, etc. 

There is an event `visibilitychange`
https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event

It could be used to remember the exact time when user leaves a tab and check if a callback was fired otherwise it needs to invoke it on user returns back.

**Comment by SupremeTechnopriest** [comment]
Can we try this again against the latest release candidate?  I think some other changes might resolve this as well. `npm i react-idle-timer@next`.

**Comment by QassemNa** [comment]
Not OP, but I tested latest release candidate multiple times in browser and it has always detected idle time, tested on chrome browser and in IOS web view app. and run the same test in the latest release it was almost always failing.

the times settings were `timeout: 1000 * 60 * 9, promptTimeout: 1000 * 60`

**Comment by denysoleksiienko** [comment]
Still not working `npm i react-idle-timer@next` on Android Chrome:
```
const timeout = 1000 * 60 * 4;
const promptTimeout = 1000 * 60 * 2;
```
```

 const [open, setOpen] = useState<boolean>(false);
 const [remaining, setRemaining] = useState<number>(0);

 const onPrompt = useCallback(() => {
    setOpen(true);
    setRemaining(promptTimeout);
 }, []);

 const onIdle = useCallback(() => {
    setOpen(false);
    setRemaining(0);
    push('/auth/signout');
 }, []);

 const { getRemainingTime, isPrompted, activate, pause } = useIdleTimer({
    timeout,
    promptTimeout,
    onPrompt,
    onIdle,
    timers: workerTimers,
 });

 const handleStillHere = useCallback(() => {
    setOpen(false);
    activate();
 }, []);

 const handleLogout = useCallback(() => {
    setOpen(false);
    setRemaining(0);
    push('/auth/signout');
 }, []);

 useEffect(() => {
   const interval = setInterval(() => {
      if (isPrompted()) {
        setRemaining(getRemainingTime());
      }
    }, 1000);
    return () => {
      clearInterval(interval);
    };
 }, []);

```

**Comment by zhangjianrencai** [comment]
@AlexHuman how do you define your workerTimers

**Comment by MinChanSike** [comment]
@zhangjianrencai 
```ts
import * as workerTimers from "worker-timers";

const {start, activate, getRemainingTime, isPrompted} = useIdleTimer({
    name: "IdleTimerTest",
    timers: workerTimers,
    ....
})
```
